### DISCUSSION TITLE
Context propagation not working with Spring Boot

### DISCUSSION THREAD

**Discussion by mNantern** [open-telemetry/opentelemetry-java-instrumentation#4330] [Q&A] [answered]
Hello 👋

I'm trying to auto-instrument my spring boot application with the java agent but unfortunately I cannot manage to obtain a working configuration for context propagation.

When I call my API at `localhost:8080/api/ping` with curl I don't see any `traceparent` header in the response.

This is how I run my application:

```
java -javaagent:path/opentelemetry-javaagent-all-1.6.2.jar \
       -Dotel.resource.attributes=service.name=backend \
       -Dotel.javaagent.experimental.thread-propagation-debugger.enabled=true \ 
       -Dotel.propagation.enabled=true \
       -Dotel.metrics.exporter=none \ 
       -jar web/build/libs/web.jar
```

And every time I have the following error in the logs:

```
i.o.i.a.i.ContextPropagationDebug   | Unexpected non-root current context found when extracting remote context!
i.o.i.a.i.ContextPropagationDebug   | It contains this span: io.opentelemetry.javaagent.instrumentation.opentelemetryapi.trace.ApplicationSpan@5890f9db
```

I've tested it with the javaagent v1.6.2 and v1.5.3.

Any idea on how I may debug my issue here? 🙏

**Comment by iNikem** [comment, marked as answer]
First, can you share a minimal reproducible test case for the error?

Second, why do you expect to have `traceparent` header in the response? That header is used to propagate span context with outgoing requests. When you call a service, a span is created to track the processing of that request and that span ends when request processing finishes. There is no span context to propagate _back_ with the response.

**Reply by mNantern** [reply]
> why do you expect to have traceparent header in the response? That header is used to propagate span context with outgoing requests. When you call a service, a span is created to track the processing of that request and that span ends when request processing finishes. There is no span context to propagate back with the response.

You are right, I was making a (false) assumption here, `traceparent` is only required when they are multiple services and I was testing with only one.

After adding a second service I can see that the context propagation is happening as expected, thank you for your answer and sorry for wasting your time 🙏.
